secret_store/src/key_server_cluster/cluster.rs

```diff
@@ -165,7 +165,7 @@ pub struct ClusterClientImpl {
 
 /// Network cluster view. It is a communication channel, required in single session.
 pub struct ClusterView {
-	core: Arc<Mutex<ClusterViewCore>>,
+	core: Arc<RwLock<ClusterViewCore>>,
```

Comment:
It doesn't break anything. I would also prefer not to mix different changes in single PR, but I guess these are from times when @c0gent tried to fix SS tests. Since this changes doesn't break anything + this PR is too old && big already, I'm OK with merging as-is.

Reply:
Could you provide some deadlocked stacks? I haven't seen any

Reply:
I was going to add that the change you suggested may have redundantly fixed the problem. Should I switch back then?

Reply:
Remember the deadlocking was intermittent and it only seemed to happen on CI, which is why I left both changes just to be sure.

Reply:
I haven't seen any deadlocked threads while working on this change, but I have seen that tokio has stopped waking up tasks. Have you seen some? If so - please share. I'm not insisting to revert this, as I said, but am warned about deadlocking.

Reply:
Hanging tests were easy reproducible locally on my laptop, but again - I haven't seen any deadlocks in SS code (related to `Mutex` -> `RwLock` changes). So I'm not sure this change is about deadlock-fix - just possible change (we could even call it optimization :) ).

Reply:
Well should we just call it an optimization and leave it be?

Reply:
Yes - let's leave as is